MDS: add MCP settings

eoigal opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of https://linear.app/a8c/issue/DSGCOM-317/mcp-adapt-flows-for-the-multi-site-dashboard#comment-236bfcba

Adds MCP (Model Context Protocol) settings pages to the Multi-site Dashboard, allowing users to configure AI assistant access to their WordPress.com account and sites. This is a feature-flagged addition that integrates with the existing MCP infrastructure.

<img width="707" height="476" alt="Screenshot 2025-12-23 at 19 17 51" src="https://github.com/user-attachments/assets/e4a2f336-746a-48f0-8426-ee2a177711a4" />
<img width="344" height="989" alt="Screenshot 2025-12-23 at 19 07 04" src="https://github.com/user-attachments/assets/05cf8a96-952b-4425-bfe3-2b7043c46e96" />
<img width="542" height="1067" alt="Screenshot 2025-12-23 at 19 06 39" src="https://github.com/user-attachments/assets/f54d254b-1c28-40d7-a5f1-f9dd75142525" />
<img width="678" height="385" alt="Screenshot 2025-12-23 at 19 04 12" src="https://github.com/user-attachments/assets/9a08a6d0-1cfd-4f87-8ca7-1c46b5301b67" />

## Proposed Changes

#### New Routes (`client/dashboard/app/router/me.tsx`)
- Added `mcpRoute` (parent route with loader for `userSettingsQuery`)
- Added `mcpIndexRoute` (main MCP settings page at `/me/mcp`)
- Added `mcpSetupRoute` (MCP client setup page at `/me/mcp/setup`)
- Routes are conditionally registered based on `mcp-settings` feature flag

#### New Components

**MCP Settings Page** (`client/dashboard/me/mcp/index.tsx`)
- AI Access master toggle to enable/disable all MCP tools
- Categorized tool list organized by display categories (Sites & Content, Account, Billing, etc.)
- Site-specific settings to disable MCP access per site
- "Configure MCP Client" button (disabled when no tools are enabled)
- Uses `useSuspenseQuery` for user settings (preloaded at route level)
- Uses `useQuery` for sites list with loading state
- Auto-save with snackbar notifications via React Query mutations

**MCP Client Setup Page** (`client/dashboard/me/mcp/setup/index.tsx`)
- Multi-client support (Claude Desktop, Cursor, VS Code, Continue, Other)
- Client-specific configuration JSON generation
- Quick setup options:
  - Claude Desktop: MCPB file download
  - Cursor: One-click install link
- Manual setup with copy-to-clipboard functionality
- Documentation links for each client
- Validation to ensure at least one tool is enabled before setup

**Category Mapping** (`client/dashboard/me/mcp/categories.ts`)
- Maps tool IDs to display categories
- Defines category order for consistent UI
- Handles uncategorized tools

#### Navigation (`client/dashboard/me/me-menu/index.tsx`)
- Added "MCP" menu item (conditionally shown based on `mcp-settings` feature flag)

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Click to the `Dashboard Live` link in https://github.com/Automattic/wp-calypso/pull/107830#issuecomment-3681715648
* Go to MCP settings `/me/mcp/` 
* Confirm works as expects and is inline with the figma design.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

eoigal committed:
MDS: add MCP settings

eoigal committed:
Organize MCP tools by categories with display headers

Add category mapping to group MCP tools under category headers
(Sites & Content, Account, Billing, etc.) for better organization.

- Add categories.ts with tool-to-category mapping
- Group and display tools by category with headers
- Update UI spacing and layout

eoigal committed:
fix mcp/setup route

eoigal committed:
Fix layout/styles on setup page

eoigal committed:
tidy up spacing in lists

arthur791004 commented on client/dashboard/app/router/me.tsx:
```diff
@@ -773,6 +774,53 @@ export const appsRoute = createRoute( {
+	getParentRoute: () => mcpRoute,
+	path: '/',
+	loader: async () => {
+		await queryClient.ensureQueryData( userSettingsQuery() );
```
nit: We can move it to `mcpRoute` so we don't need it on both `mcpIndexRoute` and `mcpSetupRoute`

eoigal replied:
Done

arthur791004 commented on client/dashboard/me/mcp/categories.ts:
```diff
@@ -0,0 +1,93 @@
+	const toolName = toolId.replace( 'wpcom-mcp/', '' );
+
+	// Sites & Content
+	if (
```
Would it be better to define a mapping object?

```js
const TOOL_CATEGORY_MAP: Record<string, string> = {
	// Sites & Content
	'user-sites-resource': DISPLAY_CATEGORIES.SITES_CONTENT,
	'user-sites': DISPLAY_CATEGORIES.SITES_CONTENT,
	'site-users': DISPLAY_CATEGORIES.SITES_CONTENT,
	'posts-search': DISPLAY_CATEGORIES.SITES_CONTENT,
	'post-get': DISPLAY_CATEGORIES.SITES_CONTENT,
	'site-comments-search': DISPLAY_CATEGORIES.SITES_CONTENT,

	// Account
	'user-profile': DISPLAY_CATEGORIES.ACCOUNT,
	'user-security': DISPLAY_CATEGORIES.ACCOUNT,
	'user-achievements': DISPLAY_CATEGORIES.ACCOUNT,

	// Billing
	'user-subscriptions': DISPLAY_CATEGORIES.BILLING,

	// Notifications
	'user-notifications': DISPLAY_CATEGORIES.NOTIFICATIONS,
	'user-notifications-inbox': DISPLAY_CATEGORIES.NOTIFICATIONS,

	// Domains & Integrations
	'user-domains': DISPLAY_CATEGORIES.DOMAINS_INTEGRATIONS,
	'user-connections': DISPLAY_CATEGORIES.DOMAINS_INTEGRATIONS,

	// Site Configuration
	'site-plugins': DISPLAY_CATEGORIES.SITE_CONFIGURATION,
	'site-settings': DISPLAY_CATEGORIES.SITE_CONFIGURATION,
	'site-statistics': DISPLAY_CATEGORIES.SITE_CONFIGURATION,

	// Developer & Testing
	'sample-prompt': DISPLAY_CATEGORIES.DEVELOPER_TESTING,
};
```

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+	const queryClient = useQueryClient();
+	const { queries } = useAppContext();
+	// @ts-expect-error - site_visibility is valid but not in type definition
+	const { data: sites = [] } = useQuery( queries.sitesQuery( { site_visibility: 'visible' } ) ) as {
```
We need to pass `include_a8c_owned` to `queries.sitesQuery`. Or we can make it optional and default to false 🙂

eoigal replied:
Updated to set `include_a8c_owned` to false.

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+		data: userSettings,
+		isLoading: isLoadingUserSettings,
+		error: userSettingsError,
+	} = useQuery( userSettingsQuery() );
```
Since we preload the user settings on the route level, we can now use `useSuspenseQuery` here to ensure the data is always available

eoigal replied:
Done!

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+	}
+
+	return (
+		<PageLayout size="small" header={ <PageHeader title={ __( 'MCP Client Setup' ) } /> }>
```
Let's add `Breadcrumbs` to this page so the user can go back to previous page easily.

```
prefix={ <Breadcrumbs length={ 2 } /> }
```

eoigal replied:
Done

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+							'MCP (Model Context Protocol) enables AI assistants to securely access and interact with your WordPress.com data. <link>Learn more.</link>'
+						),
+						{
+							link: <InlineSupportLink supportContext="mcp" />,
```
`InlineSupportLink` has default text `Learn more` for consistency so we can simply replace `<link>...</link>` with `<learnMoreLink />`

eoigal replied:
Done

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+				<Card>
+					<CardBody>
+						<VStack spacing={ 4 }>
+							<div
```
I think we can use `HStack`?

eoigal replied:
Done!

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+								style={ {
+									display: 'flex',
+									justifyContent: 'space-between',
+									alignItems: 'center',
```
The Figma shows it should align to top

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+									) }
+								/>
+								{ anyToolsEnabled && (
+									<Link to="/me/mcp/setup">
```
nit: It would be better to pass it as the `actions` props of `SectionHeader`. But there is the wrap issue here 🫠

eoigal replied:
I'm not sure worth adding as `actions` - it makes solving the wrapping issue more difficult. I've updated this to wrap better at different resolutions.

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+								checked={ anyToolsEnabled }
+								onChange={ handleMasterToggle }
+								label={
+									<Text weight="bold">
```
The Figma shows the normal font weight

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+									value={ selectedSiteId }
+									onChange={ ( value ) => setSelectedSiteId( value || '' ) }
+									label={ __( 'Select a site to disable MCP access' ) }
+									isLoading={ false }
```
Would it be better to pass the loading state of the query?

eoigal replied:
Done

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+								<PreferencesLoginSiteDropdown
+									sites={ sites }
+									value={ selectedSiteId }
+									onChange={ ( value ) => setSelectedSiteId( value || '' ) }
```
Let's align with other usages to store the value as int, and pass it to this component as string. If so, we don't need to do `parseInt` elsewhere WDTY?

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+									<Text weight="bold">
+										{ anyToolsEnabled
+											? __( 'Disable MCP Tool Access' )
+											: __( 'Enable MCP Tool Access' ) }
```
Do we need to change the label 🤔

eoigal replied:
Probably - I'm not sure the user will know (or care) what a MCP tool is - we've also updated the list of MCP tools to not mention tools below.

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+										disabled={ mutation.isPending }
+										onChange={ ( enabled ) => handleSiteToggle( selectedSiteId, enabled ) }
+										label={
+											<Text weight="bold">
```
Same

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+											<Text weight="bold">
+												{ getSiteAccountToolsEnabled( userSettings || {}, selectedSiteId )
+													? __( 'Disable MCP access for this site' )
+													: __( 'Enable MCP access for this site' ) }
```
Do we need to change the label 🤔

eoigal replied:
I've made some changes here to remove referring to MCP and MCP tools - let me know what you think.

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,371 @@
+
+								return (
+									<VStack key={ categoryName } spacing={ 4 }>
+										<Text as="h4" size={ 14 } weight={ 600 }>
```
Figma shows `font-size: 11px` and `font-weight: 500` 🙂

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+		data: userSettings,
+		isLoading: isLoadingUserSettings,
+		error: userSettingsError,
+	} = useQuery( userSettingsQuery() );
```
As mentioned, we've preloaded the query so we can use `useSuspenseQuery` to ensure the data is always available 🙂

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+						</CardBody>
+					</Card>
+
+					<div style={ { marginTop: '24px' } }>
```
Figma shows the gap between each cards is `24px` so I think we don't need `marginTop` here and below

eoigal replied:
Will remove.

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+							<VStack spacing={ 4 }>
+								<Text as="h1" size="large" weight={ 500 }>
+									{ __( 'Connect AI Assistant to WordPress.com (MCP)' ) }
+								</Text>
```
We should use SectionHeader as Figma

eoigal replied:
Done

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+						<Card>
+							<CardBody>
+								<VStack spacing={ 4 }>
+									<Text as="h1" size="large" weight={ 500 }>
```
We should use SectionHeader as Figma

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+									{ /* Quick Setup for Claude Desktop */ }
+									{ selectedMcpClient === 'claude' && (
+										<VStack spacing={ 4 }>
+											<Text as="h1" size="large" weight={ 500 }>
```
We should use SectionHeader as Figma

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+									{ /* Quick Setup for Cursor */ }
+									{ selectedMcpClient === 'cursor' && (
+										<VStack spacing={ 4 }>
+											<Text as="h1" size={ 16 } weight={ 600 }>
```
We should use SectionHeader as Figma

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+								<Text as="h1" size="large" weight={ 500 }>
+									{ __( 'Connect AI Assistant to WordPress.com (MCP)' ) }
+								</Text>
+								<Text as="p" size="medium" variant="muted">
```
Figma shows the default font size of the Text component so we don't need to specify the size here and below

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+									) }
+								</Text>
+								<VStack spacing={ 2 }>
+									<ul style={ { color: '#757575', paddingLeft: '20px', margin: '0' } }>
```
Instead of `paddingLeft`, it would be better to use `padding-inline-start`

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+										'The JSON configuration below sets up a secure connection between your AI assistant and your WordPress.com account. It works by:'
+									) }
+								</Text>
+								<VStack spacing={ 2 }>
```
I think we don't need this 🙂

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+										<li>
+											<Text as="p" size="medium" variant="muted">
+												{ __(
+													"Providing real-time access to your account's content and management features"
```
We should use `‘` instead of `'`

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+											<Text as="h1" size="large" weight={ 500 }>
+												{ __( 'Quick Setup' ) }
+											</Text>
+											<Text as="p" size="medium" variant="muted">
```
Figma shows the default font size of the Text component so we don't need to specify the size here and below

arthur791004 replied:
As Figma shows, this seems to be the description of the SectionHeader, and the gap between each other is 20px (`spacing={ 5 }`)

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+											<Text as="p" size="medium" variant="muted">
+												{ __( 'Installation steps:' ) }
+											</Text>
+											<ol style={ { color: '#757575', paddingLeft: '20px', margin: '0' } }>
```
Same

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+												) }
+											</Text>
+											<Text as="p" size="medium" variant="muted">
+												{ __( 'Installation steps:' ) }
```
We should wrap this one and list below so make the gap align with Figma 🙂

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+											</Text>
+											<Text as="p" size="medium" variant="muted">
+												{ __(
+													'For Cursor users, we provide a one-click setup option that will automatically configure the MCP server.'
```
Same. I think this is also the description of the SectionHeader, and the gap is 20px (`spacing={ 5 }`)

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+						<Card>
+							<CardBody>
+								<VStack spacing={ 4 }>
+									<Text as="h1" size="large" weight={ 500 }>
```
We should use SectionHeader here as well

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+									</VStack>
+								</VStack>
+								<VStack spacing={ 3 }>
+									<ul style={ { listStyle: 'none', padding: '0', margin: '0' } }>
```
Do we need to use ul here. I think we can remove this and just keep VStack. WDYT? 🤔

eoigal replied:
I updated to use vstack below the textarea, however I _think_ I need to use the other lists as they are in the other sections.

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+											) }
+										</li>
+									</ul>
+								</VStack>
```
I'm not sure but I noticed there is a download button on Figma

eoigal replied:
I believe that's a mistake. cc @lucasmendes-design - there is no download button in original page here - I think its from the Claude desktop `Download MCPB File` in the Quick setup section.

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+									</Text>
+
+									<VStack spacing={ 3 }>
+										<div
```
It seems it should be the label of the `TextareaControl`. But then we need to adjust the size of the copy button

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,437 @@
+												</li>
+												<li>
+													<Text as="p" size="medium" variant="muted">
+														{ __( "Follow Claude Desktop's setup instructions" ) }
```
We should use `‘` instead of `'`

lucasmendes-design commented:
@eoigal @arthur791004 do I need the sandbox to test this out? I can't open the MCP within the new MSD.

eoigal committed:
tidy up more styles

eoigal committed:
consolidate the preload into parent route

eoigal committed:
update getDisplayCategory to use mapping object

eoigal committed:
Update sites query to pass include_a8c_owned as false, and use useSuspenseQuery

eoigal committed:
more style changes

lucasmendes-design commented:
Thanks, @fushar. It worked :) My brain was repeating the mental model of openning the Calypso live link 😆

lucasmendes-design commented:
Great work @eoigal 👌🏽 

**Font-weight** 
Let's use a normal font weight here. It's too strong this way. Our DS needs a better treatment for this use case. 

<img width="787" height="257" alt="image" src="https://github.com/user-attachments/assets/703b698a-6cff-4dfa-a50a-7f4d18fb67e6" />

Same here. Let's use the normal font weight. 

<img width="724" height="233" alt="image" src="https://github.com/user-attachments/assets/2345305a-bea0-4d91-be04-f4a774d83bc0" />

**Configure MCP Client button**
I'd keep this button and just make it disabled. This way, we avoid the jump between 2 lines and 1 line of copy.

eoigal committed:
save siteID as int

eoigal committed:
tidy up phrasing around MCP tools and AI Access

eoigal committed:
update learnmorelink

eoigal committed:
use useSuspenseQuery

eoigal committed:
use site loading state

eoigal committed:
add breadcrumbs and use hstack

eoigal committed:
use vstack instead of ul

eoigal committed:
disable button if AI access is disabled

eoigal committed:
update comma

eoigal committed:
remove style

lucasmendes-design commented:
Thanks for the swift work, @eoigal. It looks great 👌🏽

I've created this linear issue so we can improve the Site-specific MCP settings once I'm back from holidays. 

https://linear.app/a8c/issue/DSGCOM-343/improve-the-site-specific-mcp-settings

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,365 @@
+									) }
+								/>
+								<VStack style={ { flexShrink: 0 } }>
+									<Link to="/me/mcp/setup">
```
We can use `RouterLinkButton` 🙂

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,365 @@
+									<VStack key={ categoryName } spacing={ 4 }>
+										<Text
+											as="h4"
+											size="x-small"
```
The Figma shows `11px` so maybe `smaller` would be better.

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+					<CardBody>
+						<SectionHeader level={ 3 } title={ __( 'Setup Required' ) } />
+						<VStack spacing={ 4 }>
+							<Text as="p" size="medium" variant="muted">
```
I think we should remove any usage of `size="medium"` as it doesn't match the styles on Figma

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+			>
+				<ComponentViewTracker eventName="calypso_dashboard_mcp_setup_view" />
+				<Card>
+					<CardBody>
```
We need to wrap the children of the `CardBody` with `VStack` as there is a gap between `SectionHeader` and other elements

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+									'MCP access defines what actions and data your MCP client can access on your account. You need to enable MCP access in the main MCP settings before configuring your client.'
+								) }
+							</Text>
+							<Link to="/me/mcp">
```
Same. We can use `RouterLinkButton` here

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+							title={ __( 'Connect AI Assistant to WordPress.com (MCP)' ) }
+						/>
+						<VStack spacing={ 4 }>
+							<Text as="p" size="medium" variant="muted">
```
Same. We should remove any usage of size="medium"

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+			<ComponentViewTracker eventName="calypso_dashboard_mcp_setup_view" />
+			<>
+				<Card>
+					<CardBody>
```
Same. We need to wrap the children of the `CardBody` with `VStack`

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+				</Card>
+
+				<Card>
+					<CardBody>
```
Same. We need to wrap the children of the `CardBody` with `VStack`

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+
+				{ ( selectedMcpClient === 'claude' || selectedMcpClient === 'cursor' ) && (
+					<Card>
+						<CardBody>
```
Same. We need to wrap the children of the `CardBody` with `VStack`

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+				) }
+
+				<Card>
+					<CardBody>
```
Same. We need to wrap the children of the `CardBody` with `VStack`

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+							{ selectedMcpClient === 'cursor' && (
+								<VStack spacing={ 4 }>
+									<Text as="h1" size={ 16 } weight={ 600 }>
+										{ __( 'Quick Setup' ) }
```
Should we remove this? Otherwise, it becomes:

<img width="679" height="219" alt="image" src="https://github.com/user-attachments/assets/c9a827f9-e64e-40f8-9edd-0949318f764a" />

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+			}
+		>
+			<ComponentViewTracker eventName="calypso_dashboard_mcp_setup_view" />
+			<>
```
Do we need this Fragment?

eoigal replied:
I think so. It gives us an indication of how many users view the setup/config instructions.

arthur791004 commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,365 @@
+		},
+		onSuccess: ( newData: any ) => {
+			// Update the cache with the new data from the API response
+			queryClient.setQueryData( userSettingsQuery().queryKey, newData as any );
```
Blocker: We should merge the `oldData`. Otherwise, the user settings only contain the mcp settings

fushar replied:
Do we even need this? We already have the invalidation in the mutation itself:

https://github.com/Automattic/wp-calypso/blob/5a8974131062e9c3935cf6d68a39eb908caf1c49/packages/api-queries/src/me-settings.ts#L14-L23

arthur791004 reviewed (commented):
Tested and works as expected. Nice job 👍 BTW I think we can enable the feature except the production.

There are some style issues, and main issues are
* The `Text` with the `medium` font size.
* The cache after updating is incorrect.
* We should use curly quote `’` instead of `'`

We can move this forward after resolving above issues 🙂

Other issues as follows can be addressed in the follow-up PR
* The gap after `SectionHeader`
* Replace `Link` and `Button` with `RouterLinkButton`
* The `x-small` font size on the category name
* Instead of paddingLeft, it would be better to use paddingInlineStart

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+									<li>
+										<Text as="p" size="medium" variant="muted">
+											{ __(
+												"Providing real-time access to your account's content and management features"
```
We should use curly quote `’` instead of `'`

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+								onChange={ () => {} } // Required prop for read-only textarea
+								readOnly
+								help={ __(
+									"Copy this configuration and paste it into your MCP client's settings."
```
We should use curly quote ’ instead of '

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+									<Text as="p" size="medium" variant="muted">
+										{ __( 'Installation steps:' ) }
+									</Text>
+									<ol style={ { color: '#757575', paddingLeft: '20px', margin: '0' } }>
```
Instead of paddingLeft, it would be better to use padding-inline-start

arthur791004 commented on client/dashboard/me/mcp/setup/index.tsx:
```diff
@@ -0,0 +1,411 @@
+								) }
+							</Text>
+							<VStack spacing={ 2 }>
+								<ul style={ { color: '#757575', paddingLeft: '20px', margin: '0' } }>
```
Instead of paddingLeft, it would be better to use padding-inline-start

StevenDufresne commented:
I mentioned this in the figma, but [sentence case should be used](https://github.com/Automattic/wp-calypso/blob/trunk/client/dashboard/docs/typography-and-copy.md) for almost everything. Can we update the card titles and labels to align?

eoigal committed:
use RouterLinkButton

eoigal committed:
Update MCP site settings to support multi-select with FormTokenField

Replace single site dropdown with FormTokenField for multi-select functionality.
Users can now select multiple sites (displayed as removable chips) and control
AI access for all selected sites with a single toggle. Fixes duplicate site
name handling by using unique site IDs in token values.

eoigal committed:
Tidy up some page layout & style issues

eoigal committed:
Remove redundant cache update in MCP settings mutation

userSettingsMutation() already handles cache updates. Remove custom
onSuccess handler and unused useQueryClient import.

arthur791004 reviewed (approved):
Nice job 👍

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/31456391

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @eoigal for including a screenshot in the description! This is really helpful for our translators.

emilyaudela commented on client/dashboard/me/mcp/index.tsx:
```diff
@@ -0,0 +1,396 @@
+									level={ 3 }
+									title={ __( 'AI Access' ) }
+									description={ __(
+										'Control what AI assistants can access your WordPress.com account and sites.'
```
The text is unclear. Is it about about controlling “which” AI assistants can access stuff, or rather “what” the AI assistants can access IN accounts and sites?

If it's the former, it should be "Control which AI assistants can access..."

If it's the latter, it should include a preposition: "Control what AI assistants can access `in` your WordPress.com account and sites."
